mrp_bom_version/models/mrp_bom.py

```diff
@@ -0,0 +1,178 @@
+# (c) 2015 Oihane Crucelaegui - AvanzOSC
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from odoo import models, fields, api
+from odoo.tools import config
+
+
+class MrpBom(models.Model):
+    _inherit = "mrp.bom"
+
+    @api.multi
+    def _track_subtype(self, init_values):
+        self.ensure_one()
+        if "state" in init_values and self.state == "active":
+            return "mrp_bom_version.mt_active"
+        return super(MrpBom, self)._track_subtype(init_values)
+
+    @api.multi
+    def _compute_old_versions(self):
+        self.ensure_one()
+        parent = self.parent_bom
+        old_version = self.env["mrp.bom"]
+        while parent:
+            old_version += parent
+            parent = parent.parent_bom
+        self.old_versions = old_version
+
+    def _default_active(self):
+        """Needed for preserving normal flow when testing other modules."""
+        res = False
+        if config["test_enable"]:
+            res = not bool(self.env.context.get("test_mrp_bom_version"))
+        return res
+
+    def _default_state(self):
+        """Needed for preserving normal flow when testing other modules."""
+        res = "draft"
+        if config["test_enable"] and not self.env.context.get(
+            "test_mrp_bom_version"
+        ):
+            res = "active"
+        return res
+
+    active = fields.Boolean(
+        default=_default_active,
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+    )
+    historical_date = fields.Date(string="Historical Date", readonly=True)
+    state = fields.Selection(
+        selection=[
+            ("draft", "Draft"),
+            ("active", "Active"),
+            ("historical", "Historical"),
+        ],
+        string="State",
+        track_visibility="onchange",
+        index=True,
+        readonly=True,
+        default=_default_state,
+        copy=False,
+    )
+    product_tmpl_id = fields.Many2one(
+        readonly=True, states={"draft": [("readonly", False)]}
+    )
+    product_id = fields.Many2one(
+        readonly=True, states={"draft": [("readonly", False)]}
+    )
+    product_qty = fields.Float(
+        readonly=True, states={"draft": [("readonly", False)]}
+    )
+    code = fields.Char(states={"historical": [("readonly", True)]})
```

Comment:
it's cosmetic but fields description aren't all formatted the same way.

Reply:
`black` formats them according if they fit into one line or not given the line length limit.